distributed/worker.py

```diff
@@ -2719,10 +2724,7 @@ def get_worker() -> Worker:
     try:
         return thread_state.execution_state["worker"]
     except AttributeError:
-        try:
-            return first(w for w in Worker._instances if w.status in WORKER_ANY_RUNNING)
-        except StopIteration:
-            raise ValueError("No workers found")
```

Comment:
I get the appeal of this but this can throw off some of our test logic and tests suggest that certain functionality would work which in reality doesn't.

Reply:
i.e. due to our async tests, we are likely to get a worker just because in the same process an instance already exists. However, in reality there is no valid worker in the context, i.e. if the test example would be executed on a "real" cluster, it would fail